mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -146,11 +164,13 @@ class BinaryClassificationMetrics @Since("1.3.0") (
   private lazy val (
     cumulativeCounts: RDD[(Double, BinaryLabelCounter)],
     confusions: RDD[(Double, BinaryConfusionMatrix)]) = {
-    // Create a bin for each distinct score value, count positives and negatives within each bin,
-    // and then sort by score values in descending order.
-    val counts = scoreAndLabels.combineByKey(
-      createCombiner = (label: Double) => new BinaryLabelCounter(0L, 0L) += label,
-      mergeValue = (c: BinaryLabelCounter, label: Double) => c += label,
+    // Create a bin for each distinct score value, count weighted positives and
+    // negatives within each bin, and then sort by score values in descending order.
+    val counts = scoreLabelsWeight.combineByKey(
+      createCombiner = (labelAndWeight: (Double, Double)) =>
+        new BinaryLabelCounter(0.0, 0.0) += (labelAndWeight._1, labelAndWeight._2),
+      mergeValue = (c: BinaryLabelCounter, labelAndWeight: (Double, Double)) =>
+        c += (labelAndWeight._1, labelAndWeight._2),
```

Comment:
Isn't this the same as `c += labelAndWeight`?

Reply:
I would think so but for some reason it didn't like that:
[error] /home/ilya/ilmat-spark/mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala:173: overloaded method value += with alternatives:
[error]   (other: org.apache.spark.mllib.evaluation.binary.BinaryLabelCounter)org.apache.spark.mllib.evaluation.binary.BinaryLabelCounter <and>
[error]   (label: Double)org.apache.spark.mllib.evaluation.binary.BinaryLabelCounter
[error]  cannot be applied to ((Double, Double))
[error]         c += labelAndWeight,
[error]           ^

Reply:
Hm, confusing, because the current code is also passing a (Double, Double), but also, there doesn't seem to be a += method for (Double, Double) anyway. I'm not sure why this works! maybe some implicit conversion happening? weird.

Reply:
oh, there is a += operator overloaded for (Double, Double), see BinaryLabelCounter.scala (I added it in this PR - in addition to the += Label operator overload that already existed there).  Maybe having an operator overload that works on a tuple is confusing to add?  Perhaps I need to add an explicit method instead - but this seems cleaner and more in line with the existing += operator overload that takes the label as a Double.